src/xercesc/validators/common/DFAContentModel.cpp

```diff
@@ -661,8 +662,15 @@ void DFAContentModel::buildDFA(ContentSpecNode* const curNode)
     //  in the fLeafCount member.
     //
     fLeafCount=countLeafNodes(curNode);
+    // Avoid integer overflow in below fLeafCount++ increment
+    if (fLeafCount > std::numeric_limits<unsigned int>::max() - 1)
+        throw OutOfMemoryException();
     fEOCPos = fLeafCount++;
 
+    // Avoid integer overflow in below memory allocatoin
+    if (fLeafCount > std::numeric_limits<size_t>::max() / sizeof(CMLeaf*))
```

Comment:
Should this also be `XMLSize_t`?

Would it be possible to add brackets around the division for clarity?

Reply:
> Would it be possible to add brackets around the division for clarity?

this is the best way I can think of that doesn't rely on doing overflow arithmetic... As it is unsigned type, we could do overflow arithmetic as it is well defined in C/C++, but I tend to fuzz software with -fsanitize=unsigned-integer-overflow because in practice > 90% unsigned overflows that occur are actually bugs

Reply:
I just meant `fLeafCount > (std::numeric_limits<size_t>::max() / sizeof(CMLeaf*))` so that the operator precedence is explicit (and the same for below).

Reply:
ah sorry for some reason I read your sentence as "would it be possible to *avoid* the division"... Brackets added